lib/sdt_task.bpf.c

```diff
@@ -269,7 +271,7 @@ static SDT_TASK_FN_ATTRS int sdt_pool_set_size(struct sdt_task_pool __arena *poo
 }
 
 /* initialize the whole thing, maybe misnomer */
-static SDT_TASK_FN_ATTRS int sdt_task_init(__u64 data_size)
+__hidden SDT_TASK_FN_ATTRS int sdt_task_init(__u64 data_size)
```

Comment:
I am confused with the following code. The code seems to return with the lock on failure (-ENOMEM) not on success. Am I missing something? Also, in case of error (or success), why is it reasonable to hold the lock?

```.c
	/* On success, returns with the lock taken. */
	ret = sdt_alloc_attempt(prealloc_stack);
```

Reply:
The call unlocks the spinlock on all error paths this is also largely enforced by the verifier. The reason ```sdt_alloc_attempt```  keepsthe lock on success is because success means the preallocation stack has enough data to satisfy all the memory needs of the ```sdt_task_allloc``` without having to make a sleepable arena allocation. Dropping the lock would invalidate this guarantee.

Reply:
One question. Many scx schedulers use `cpu_ctx`. Do you plan to extend the current task slab allocator to a bit more general slab allocator covering both task and CPU?

Reply:
Yes, the plan is to evolve the allocator towards being able to define multiple suballocators. It should be a matter of adding an extra layer of indirection in the code, but I went for a simpler approach for starters.